sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/ReceiptExtensions.java

```diff
@@ -0,0 +1,133 @@
+                case "Subtotal":
+                    subtotal = (FormField<Float>) fieldValue;
+                    break;
+                case "Tax":
+                    tax = (FormField<Float>) fieldValue;
+                    break;
+                case "Tip":
+                    tip = (FormField<Float>) fieldValue;
+                    break;
+                case "Total":
+                    total = (FormField<Float>) fieldValue;
+                    break;
+                case "TransactionDate":
+                    transactionDate = (FormField<LocalDate>) fieldValue;
+                    break;
+                case "TransactionTime":
+                    transactionTime = (FormField<String>) fieldValue;
+                    break;
+                case "Items":
+                    receiptItems = toReceiptItems(fieldValue);
+                    break;
+                default:
+                    break;
+            }
+        }
+        return new USReceipt(receipt.getReceiptLocale(), receipt.getRecognizedForm(), receiptItems, receiptType,
+            merchantName, merchantAddress, merchantPhoneNumber, subtotal, tax, tip, total, transactionDate,
+            transactionTime);
+    }
+
+    /**
+     * Helper method to convert the service level
+     * {@link com.azure.ai.formrecognizer.implementation.models.FieldValue#getValueArray() value items}
+     * to SDK level {@link USReceiptItem receipt items}.
+     *
+     * @param fieldValueItems The strongly typed field values.
+     *
+     * @return A list of {@link USReceiptItem}.
+     */
+    private static List<USReceiptItem> toReceiptItems(FormField<?> fieldValueItems) {
+        List<FormField<?>> fieldValueArray = (List<FormField<?>>) fieldValueItems.getFieldValue();
+        List<USReceiptItem> receiptItemList = new ArrayList<>();
+        FormField<String> name = null;
+        FormField<Float> quantity = null;
+        FormField<Float> price = null;
+        FormField<Float> totalPrice = null;
+        USReceiptItem receiptItem = null;
+
+        for (FormField<?> eachFieldValue : fieldValueArray) {
+            Map<String, FormField<?>> objectValue = ((Map<String, FormField<?>>) (eachFieldValue.getFieldValue()));
+            for (Map.Entry<String, FormField<?>> entry : objectValue.entrySet()) {
+                String key = entry.getKey();
+                if (QUANTITY.toString().equals(key)) {
+                    quantity = (FormField<Float>) entry.getValue();
+                } else if (NAME.toString().equals(key)) {
+                    name = (FormField<String>) entry.getValue();
+                } else if (PRICE.toString().equals(key)) {
+                    price = (FormField<Float>) entry.getValue();
+                } else if (TOTAL_PRICE.toString().equals(key)) {
+                    totalPrice = (FormField<Float>) entry.getValue();
+                }
+                receiptItem = new USReceiptItem(name, quantity, price, totalPrice);
```

Comment:
Should move the instantiation of `USReceiptItem` outside of the inner for loop, this will construct it when the entire `entrySet` has been processed and will churn a lot of short lived objects.

Reply:
updated.